shared/chat/conversation/list-area/normal/index.desktop.js

```diff
@@ -58,22 +65,26 @@ class Thread extends React.Component<Props> {
   componentWillReceiveProps(nextProps: Props) {
     if (this.props.conversationIDKey !== nextProps.conversationIDKey) {
       this._cellCache.clearAll()
-      this._isLockedToBottom = true
+      this.setState({isLockedToBottom: true})
     }
 
     if (this.props.messageOrdinals.size !== nextProps.messageOrdinals.size) {
-      // Just the top items can change size
-      this._cellCache.clear(0, 0)
-      this._cellCache.clear(1, 0)
+      // Force the grid to throw away its local index based cache. There might be a lighterway to do this but
+      // this seems to fix the overlap problem. The cellCache has correct values inside it but the list itself has
+      // another cache from row -> style which is out of sync
+      this._cellCache.clearAll()
+      this._list && this._list.Grid && this._list.recomputeRowHeights(0)
     }
   }
 
-  _updateBottomLock = (clientHeight: number, scrollHeight: number, scrollTop: number) => {
+  _updateBottomLock = throttle((clientHeight: number, scrollHeight: number, scrollTop: number) => {
     // meaningless otherwise
     if (clientHeight) {
-      this._isLockedToBottom = scrollTop + clientHeight >= scrollHeight - lockedToBottomSlop
+      this.setState({
```

Comment:
can you avoid calling this if `isLockedTobottom` hasn't changed? This re-renders `onScroll` basically